SimpleWAWS/Code/CsmExtensions/CsmResourceGroupExtensions.cs

```diff
@@ -83,26 +89,17 @@ public static partial class CsmManager
             return resourceGroup;
         }
 
-        public static async Task<ResourceGroup> LoadJenkinsResources(this ResourceGroup resourceGroup,
-            CsmWrapper<CsmJenkinsResource> jenkinsResources = null, bool load = false)
+        public static async Task<ResourceGroup> LoadJenkinsResources(this ResourceGroup resourceGroup, CsmWrapper<CsmJenkinsResource> jenkinsResources = null)
         {
-            if (load)
+            if (jenkinsResources == null && !string.IsNullOrEmpty(resourceGroup.UserId))
```

Comment:
The only thing this loads is the IP address, and it is only expected to be there if the template was deployed to the resourcegroup. If its not deployed i.e  no UserId assigned  , then we dont need to check for the IP address .